java-checks/src/main/java/org/sonar/java/filters/LombokFilter.java

```diff
@@ -116,4 +121,42 @@ private static boolean usesAnnotation(ClassTree classTree, List<String> annotati
     }
     return false;
   }
+
+  private boolean generatesPrivateConstructor(ClassTree classTree) {
+    if (usesAnnotation(classTree, GENERATE_PRIVATE_CONSTRUCTOR)) {
+      return true;
+    }
+    SymbolMetadata metadata = classTree.symbol().metadata();
+    for (String annotation : GENERATE_CONSTRUCTOR) {
+      if (metadata.isAnnotatedWith(annotation) && generatesPrivateAccess(metadata, annotation)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private static boolean generatesPrivateAccess(SymbolMetadata metadata, String annotation) {
+    for (SymbolMetadata.AnnotationValue annotationValue : metadata.valuesForAnnotation(annotation)) {
+      if (Objects.equals(annotationValue.name(), "access") && Objects.equals(getAccessLevelValue(annotationValue.value()), "PRIVATE")) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  @Nullable
+  private static String getAccessLevelValue(Object value) {
+    String retentionValue = null;
```

Comment:
This is not a `retentionValue`... that's the Access level. Rename the variable or simply drop it to return immediately the value when getting it.

`RetentionValue` is variable name from `java.lang.annotation.Retention` (copy-pasted from ReflectionOnNonRuntimeAnnotationCheck` ?).